src/framework/actions/turn.ts

```diff
@@ -1,13 +1,120 @@
-import { GameState } from "../types";
+import { CardID, GameState } from "../types";
+import { discardAll, drawCard } from "./hand";
+import { MAX_GOLD } from "../constants";
+
+const handleReadyPhase = ($: GameState): void => {
+  if ($.turnPhase === "READY") {
+    const P = $.players[$.activePlayer];
+
+    P.hasShuffledThisTurn = false;
+    P.hasBuiltWorkerThisTurn = false;
+
+    P.patrol.squadLeader = null;
+    P.patrol.elite = null;
+    P.patrol.scavenger = null;
+    P.patrol.technician = null;
+    P.patrol.lookout = null;
+
+    for (const I of Object.values($.instances)) {
+      if (I.owner === $.activePlayer) {
+        I.arrivalFatigue = false;
+
+        I.readyState = I.readyState === "DISABLED" ? "EXHAUSTED" : "READY";
+
+        // Make dead heroes available
+      }
+    }
+
+    $.turnPhase = "UPKEEP";
+
+    handleUpkeepPhase($);
+  } else {
+    throw new Error(`not in ready phase (${$.turnPhase})`);
+  }
+};
+
+const handleUpkeepPhase = ($: GameState): void => {
+  if ($.turnPhase === "UPKEEP") {
+    const P = $.players[$.activePlayer];
+
+    P.gold = Math.min(P.gold + P.workers, MAX_GOLD);
+
+    for (const I of Object.values($.instances)) {
+      if (I.owner === $.activePlayer) {
+        // Run any upkeep actions
+      }
+    }
```

Comment:
```suggestion
```

I would honestly just remove this loop for now, since we will probably write some event dispatch helper in the future anyway (to handle all the types of events that can happen)